code/game/objects/machinery/autodoc.dm

```diff
@@ -1041,176 +1038,171 @@
-
-			if(href_list["broken"])
-				for(var/datum/limb/L in connected.occupant.limbs)
-					if(L)
-						if(L.limb_status & LIMB_BROKEN)
-							N.fields["autodoc_manual"] += create_autodoc_surgery(L,LIMB_SURGERY,ADSURGERY_BROKEN)
+	if(connected.occupant && ishuman(connected.occupant))
+		// manual surgery handling
+		var/datum/data/record/N = null
+		for(var/datum/data/record/R in GLOB.datacore.medical)
+			if (R.fields["name"] == connected.occupant.real_name)
+				N = R
+		if(isnull(N))
+			N = create_medical_record(connected.occupant)
+
+		var/needed = 0 // this is to stop someone just choosing everything
+		if(href_list["brute"])
+			N.fields["autodoc_manual"] += create_autodoc_surgery(null,EXTERNAL_SURGERY,ADSURGERY_BRUTE)
+
+		if(href_list["burn"])
+			N.fields["autodoc_manual"] += create_autodoc_surgery(null,EXTERNAL_SURGERY,ADSURGERY_BURN)
+
+		if(href_list["toxin"])
+			N.fields["autodoc_manual"] += create_autodoc_surgery(null,EXTERNAL_SURGERY,ADSURGERY_TOXIN)
+
+		if(href_list["dialysis"])
+			N.fields["autodoc_manual"] += create_autodoc_surgery(null,EXTERNAL_SURGERY,ADSURGERY_DIALYSIS)
+
+		if(href_list["blood"])
+			N.fields["autodoc_manual"] += create_autodoc_surgery(null,EXTERNAL_SURGERY,ADSURGERY_BLOOD)
+
+		if(href_list["organgerms"])
+			N.fields["autodoc_manual"] += create_autodoc_surgery(null,ORGAN_SURGERY,ADSURGERY_GERMS)
+
+		if(href_list["eyes"])
+			N.fields["autodoc_manual"] += create_autodoc_surgery(null,ORGAN_SURGERY,ADSURGERY_EYES,0,connected.occupant.internal_organs_by_name["eyes"])
+
+		if(href_list["organdamage"])
+			for(var/datum/limb/L in connected.occupant.limbs)
+				if(L)
+					for(var/datum/internal_organ/I in L.internal_organs)
+						if(I.robotic == ORGAN_ASSISTED||I.robotic == ORGAN_ROBOT)
+							// we can't deal with these
+							continue
+						if(I.damage > 0)
+							N.fields["autodoc_manual"] += create_autodoc_surgery(L,ORGAN_SURGERY,ADSURGERY_DAMAGE,0,I)
 							needed++
-				if(!needed)
-					N.fields["autodoc_manual"] += create_autodoc_surgery(null,LIMB_SURGERY,ADSURGERY_BROKEN,1)
-				updateUsrDialog()
-
-			if(href_list["missing"])
-				for(var/datum/limb/L in connected.occupant.limbs)
-					if(L)
-						if(L.limb_status & LIMB_DESTROYED)
-							if(!(L.parent.limb_status & LIMB_DESTROYED) && L.body_part != HEAD)
-								N.fields["autodoc_manual"] += create_autodoc_surgery(L,LIMB_SURGERY,ADSURGERY_MISSING)
-								needed++
-				if(!needed)
-					N.fields["autodoc_manual"] += create_autodoc_surgery(null,LIMB_SURGERY,ADSURGERY_MISSING,1)
-				updateUsrDialog()
-
-			if(href_list["necro"])
-				for(var/datum/limb/L in connected.occupant.limbs)
-					if(L)
-						if(L.limb_status & LIMB_NECROTIZED)
-							N.fields["autodoc_manual"] += create_autodoc_surgery(L,LIMB_SURGERY,ADSURGERY_NECRO)
+			if(!needed)
+				N.fields["autodoc_manual"] += create_autodoc_surgery(null,ORGAN_SURGERY,ADSURGERY_DAMAGE,1)
+
+		if(href_list["internal"])
+			for(var/datum/limb/L in connected.occupant.limbs)
+				if(L)
```

Comment:
Check and typecheck unnecessary,

Reply:
Same, just reduced indentation. But I'll fix it regardless.